src/psyclone/psyir/transformations/extract_trans.py

```diff
@@ -77,9 +77,53 @@ class ExtractTrans(PSyDataTrans):
                            HaloExchange, GlobalSum)
 
     def __init__(self, node_class=ExtractNode):
+        # This function is only here to document its interface and default.
         super(ExtractTrans, self).__init__(node_class=node_class)
 
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def determine_postfix(input_list, output_list, postfix="_post"):
+        '''
+        This function prevents any name clashes that can occur when adding
+        the postfix to output variable names. For example, if there is an
+        output variable 'a', the driver (and the output file) will contain
+        two variables: 'a' and 'a_post'. But if there is also another variable
+        called 'a_post', a name clash would occur (two identical keys in the
+        output file, and two identical local variables in the driver). In
+        order to avoid this, the suffix 'post' is changed (to 'post0',
+        'post1', ...) until any name clashes are avoided. This works for
+        structured and non-structured types.
+
+        :param input_list: list of all input variables.
+        :type input_list: list of :py:class:`psyclone.core.Signature`
+        :param output_list: list of all output variables.
+        :type output_list: list of :py:class:`psyclone.core.Signature`
+        :param str postfix: the postfix to append to each output variable.
+
+        :returns: a postfix that can be added to each output variable without
+            generating a name clash.
+        :rtype: str
+
+        '''
+        suffix = ""
+        # Create the a set of all input and output variables (to avoid
+        # checking input+output variables more than once)
+        all_vars = set(input_list) | set(output_list)
+        all_vars = set(input_list)
```

Comment:
One of these two lines must be wrong :-) Also implies we have a test missing?

Reply:
Oh gee, I remember now ... I got a headache, and decided to give it a break, and then got interrupted and forgot. So, now I have a headache again, but sorted it out.

For now, the code just checking `input_list` is correct: if there are two output only variables `out` and `out_post`, it looks like we have a conflict since the first variable will create a variable called `out_post`. But similarly, the second variable will then use `out_post_post`, and it is actually all fine. But I still don't like this, so I am using the first approach to make sure to generate unique names across input- and output-variables to be more defensive. Tests for these cases have also been added.

Reply:
After more headache I realised that my previous comment was not correct, it is already required to check for output variables as well. I've updated the comments, and tests are already in place.